src/Producer.php

```diff
@@ -111,21 +103,21 @@ public function __construct(Connection $connection, $exchangeName, $exchangeType
      * Publish a message to a specified exchange.
      *
      * @param string $data
+     * @param string $routingKey    The routing key used by the exchange to route messages to bounded queues.
      */
-    public function publish($data)
+    public function publish($data, $routingKey = null)
```

Comment:
- use `send` for producer, and `publish` for pub-sub only.
- switch the order of the parameters and have data be optional. When publishing, one might only want to publish an event occurring without providing any data.